crates/primitives/src/transaction/eip4844.rs

```diff
@@ -116,3 +150,48 @@ impl TxEip4844 {
         mem::size_of::<u128>() // max_fee_per_data_gas
     }
 }
+
+/// This represents a set of blobs, and its corresponding commitments and proofs.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct BlobTransactionSidecar {
+    /// The blob data.
+    pub blobs: Vec<Blob>,
+    /// The blob commitments.
+    pub commitments: Vec<Bytes48>,
+    /// The blob proofs.
+    pub proofs: Vec<Bytes48>,
+}
+
+impl BlobTransactionSidecar {
```

Comment:
would a function that ensures all vecs have same length be useful?

https://github.com/ethereum/go-ethereum/blob/6e934f40f9a7edd58107725d9d766770bc128775/core/txpool/validation.go#L119-L130

I don't think we should enforce this on rlp level though

Reply:
this happens in the `c-kzg` rust bindings internally (`KzgProof::verify_blob_kzg_proof_batch`), geth does this because they use a loop over `kzg4844.VerifyBlobProof`, we could add one as a sanity check

checks in `c-kzg`:
https://github.com/ethereum/c-kzg-4844/blob/4f2d632cd86ac1fe8c3175af7e34f9cca334d308/bindings/rust/src/bindings/mod.rs#L377-L390